feat: Notes Page Redesign with Zustand State Management and UX Improvements

andycaste2004 opened the pull request:
Key Features Implemented:

**Auto-Save Functionality**
- Notes save automatically 2 seconds after you stop typing
- Visual indicator shows "Saving..." or "All changes saved"
- No more manual save button needed
- New notes are created instantly in the database

**UI Modernization**
- Clean white backgrounds (Apple Notes style)
- Modern rounded buttons with smooth hover effects
- Improved sidebar with better spacing and typography
- Cleaner note editor layout
- Better visual hierarchy throughout

**Sidebar Improvements**
- Added refresh mechanism
- Changed default tab to "Unpublished" for better UX
- Removed confusing duplicate "Add Note" button
- Better note list styling with improved readability

Current Bugs
- There´s an issue with the "archive" button that is not working as expected in the UI.
- Still some issues while creating a new note, ex: text editor doesn´t refresh after creating new note and some delay issues.

Bonus:
- Finally Added password strength indicator component for signup page.

InfinityBowman reviewed (commented):
Something mentioned with our meeting with adam is to make the "add note" button absolutely/fixed positioned in the bottom left so it is never hidden by notes or scroll position, it should always be visible and clickable. make sure that when the user has lots of notes, they are able to scroll down and the add note button shouldnt interfere with them clicking on a low down note.

InfinityBowman reviewed (changes requested):
When clicking the location button, user should be able to press 'Escape', or click outside to close the menu. Remove word "Notes" at top of sidebar. 
<img width="241" alt="image" src="https://github.com/user-attachments/assets/f47705aa-517f-4d0a-b6e6-f6b9c31e5e33" />
Set min width to prevent the above item from clipping when resizing the sidebar.

Reduce spacing between generate tags button like below
<img width="589" alt="image" src="https://github.com/user-attachments/assets/e05329f6-2381-4f55-a615-09edf4d19c3e" />

Make sure we can scroll with large content
<img width="410" alt="image" src="https://github.com/user-attachments/assets/1af5e5a1-1821-4dd4-b678-158ac80f60e5" />

Get rid of this thing lol:
<img width="386" height="143" alt="Screenshot 2025-10-27 at 4 39 57 PM" src="https://github.com/user-attachments/assets/986a6c4f-3e80-41c1-b0f1-9c86be7a6a4e" />

When I click anywhere in here, it should focus the text area. Can probably be done with two query selectors on mount that add a click listener and call .focus on the text element
<img width="534" alt="image" src="https://github.com/user-attachments/assets/0a5ba519-fdc7-490b-b00a-18f1114b21fc" />

When clicking on a note, the text is not being loaded into the text editor, I've selected an element with content but the editor is blank:
<img width="453" alt="image" src="https://github.com/user-attachments/assets/76b130ed-646c-48a6-ac2d-be8b744371c0" />

andycaste2004 commented:
Summary:

- Eliminated Blank Note Spam - Removed auto-click observer that was creating unwanted empty notes on every page load; implemented deferred creation pattern (notes save only after typing content)
- Intelligent Autosave with Dirty Tracking - Added snapshot-based change detection to prevent unnecessary database writes; only saves when content actually changes and is non-blank (reduced from 2s to 500ms delay for better UX)
- Fixed Archived Notes Visibility - Added double-layer filtering (!note.isArchived) in sidebar search and filter functions to prevent deleted notes from appearing
- Removed Unnecessary Refetching - Replaced full API refetches with direct Zustand store updates after saves, reducing API calls
- Resizable Panel Persistence - Added so sidebar width persists across page refreshes; increased default width from 20% to 26%
- Empty State UX - Added contextual empty states with icons and helpful messages for "no notes" vs "no search results" scenarios
- Performance Optimization - Fixed sidebar refetch dependency issue that could cause multiple unnecessary renders
